Jenkinsfile.release

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2017-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@Library(['private-pipeline-library', 'jenkins-shared']) _
+
+import com.sonatype.jenkins.shared.Expectation
+
+properties([
+  parameters([
+    run(name: 'releaseBuild',
+        filter: 'SUCCESSFUL',
+        projectName: 'insight/insight-brain/release',
+        description: 'The latest release of IQ Server to build the docker image')
+  ])
+])
+
+String imageName = 'sonatype/nexus-iq-server',
+String version = ''
+String checksum = ''
+
+dockerizedBuildPipeline(
+  deployBranch: 'main',
+  prepare: {
+    githubStatusUpdate('pending')
+    version = getVersionFromBuildName(env.releaseBuild_NAME)
+    checksum = readBuildArtifact('insight/insight-brain/release', env.releaseBuild_NUMBER, "artifacts/nexus-iq-server-${version}-bundle.tar.gz.sha256").trim()
+    updateIQServerVersionAndChecksum(version, checksum)
+    commitAndPushChanges(version)
+  },
+  setVersion: {
+    env['VERSION'] = version.split('-')[0]
+  },
+  buildAndTest: {
+    currentBuild.displayName = "#${currentBuild.id} ${imageName}-${env.VERSION}"
+    // TODO add tests
+  },
+  testResults: ['**/validate-expectations-results.xml'],
+  lint: {
+    hadolint(['Dockerfile'])
+  },
+  vulnerabilityScan: {
+    nexusPolicyEvaluation(
+      iqApplication: 'docker-nexus-iq-server',
+      iqScanPatterns: [[scanPattern: "container:${env.DOCKER_IMAGE_ID}"]],
+      iqStage: 'release')
+  },
+  deploy: {
+    withSonatypeDockerRegistry() {
+      sh """docker tag $DOCKER_IMAGE_ID ${sonatypeDockerRegistryId()}/${imageName}-${env.BUILD_NUMBER}
+            docker push ${sonatypeDockerRegistryId()}/${imageName}-${env.BUILD_NUMBER}
+            docker rmi ${sonatypeDockerRegistryId()}/${imageName}-${env.BUILD_NUMBER}"""
```

Comment:
This will deploy to our Sonatype Internal Registry. @tneer What needs to be done so we deploy to docker hub instead? Maybe a new shared library e.g. withDockerHub which authenticates against docker hub. The current code is using this [credential](https://jenkins.ci.sonatype.dev/job/insight/job/insight-brain/job/docker/credentials/store/folder/domain/_/credential/sonatype-password/).

Reply:
@shaikhu I think we need to just copy over and clean up all that stuff from the `pushImages` stage in the original Jenkinsfile.  Most of that is specific to this set of docker images so doesn't really feel like a shared library function to me.

Reply:
> @shaikhu I think we need to just copy over and clean up all that stuff from the `pushImages` stage in the original Jenkinsfile. Most of that is specific to this set of docker images so doesn't really feel like a shared library function to me.

:+1: a83d1f21